ioengines.c

```diff
@@ -311,6 +313,38 @@ enum fio_q_status td_io_queue(struct thread_data *td, struct io_u *io_u)
 					sizeof(io_u->issue_time));
 	}
 
+	if (fio_option_is_set(o, ioprio) ||
+	    fio_option_is_set(o, ioprio_class)) {
+
+	    if ( io_u->ddir == DDIR_READ) {
+	        if (o->prio_percent) {
+	            if ((rand()%100 >= o->prio_percent) == 0) {
```

Comment:
rand() % 100 is NOT a good way to check for this. I think even the man page states that pretty clearly. This needs to be a dedicated random state so that things are manageable and repeatable (if desired). Also use rand_between() for this.

Reply:
I changed the rand() call to rand_between() including adding a new random state to the thread data structure. I squashed my update into the existing commit, but I see doing so marked your comment as outdated. In the future, should I be waiting to squash commits until the patch is more or less finalized?

Reply:
I think what you did was fine - if you touch the area where a comment is then it's going to become marked as outdated which is the expected (and desired) behaviour.